Orange/widgets/data/owcolor.py

```diff
@@ -350,6 +350,7 @@ def set_data(self, data):
                                         self._create_proxies(domain.metas))
             self.openContext(data)
             self.data = data.transform(domain)
+            self.data.domain = domain
```

Comment:
And if anyone wishes to play around, here is the minimal example of the problem, after the change introduced in this PR:
```py
import Orange
from Orange.data import Table


def _create_proxies(variables):
    part_vars = []
    for var in variables:
        if var.is_discrete or var.is_continuous:
            var = var.make_proxy()
            if var.is_discrete:
                var.values = var.values[:]
        part_vars.append(var)
    return part_vars


data = Table('iris')
domain = data.domain
domain = Orange.data.Domain(_create_proxies(domain.attributes),
                            _create_proxies(domain.class_vars),
                            _create_proxies(domain.metas))

new_data = data.transform(domain)
print(new_data)
```

The `new_data` contains all missing values since variables from the destination domain are not in the source domain and compute values are not set.